validator/src/commands/run/args/blockstore_options.rs

```diff
@@ -0,0 +1,255 @@
+use {
+    crate::{
+        cli::thread_args::{RocksdbCompactionThreadsArg, RocksdbFlushThreadsArg, ThreadArg},
+        commands::{FromClapArgMatches, Result},
+    },
+    clap::{value_t, ArgMatches},
+    solana_ledger::blockstore_options::{
+        AccessType, BlockstoreCompressionType, BlockstoreOptions, BlockstoreRecoveryMode,
+        LedgerColumnOptions,
+    },
+    std::num::NonZeroUsize,
+};
+
+impl FromClapArgMatches for BlockstoreOptions {
+    fn from_clap_arg_match(matches: &ArgMatches) -> Result<Self> {
+        let recovery_mode = matches
+            .value_of("wal_recovery_mode")
+            .map(BlockstoreRecoveryMode::from);
+
+        let column_options = LedgerColumnOptions {
+            compression_type: match matches.value_of("rocksdb_ledger_compression") {
+                None => BlockstoreCompressionType::default(),
+                Some(ledger_compression_string) => match ledger_compression_string {
+                    "none" => BlockstoreCompressionType::None,
+                    "snappy" => BlockstoreCompressionType::Snappy,
+                    "lz4" => BlockstoreCompressionType::Lz4,
+                    "zlib" => BlockstoreCompressionType::Zlib,
+                    _ => {
+                        return Err(crate::commands::Error::Dynamic(
+                            Box::<dyn std::error::Error>::from(format!(
+                                "Unsupported ledger_compression: {ledger_compression_string}"
+                            )),
+                        ));
+                    }
+                },
+            },
+            rocks_perf_sample_interval: value_t!(matches, "rocksdb_perf_sample_interval", usize)
+                .map_err(|err| {
+                    Box::<dyn std::error::Error>::from(format!(
+                        "failed to parse rocksdb_perf_sample_interval: {err}"
+                    ))
+                })?,
+        };
+
+        let rocksdb_compaction_threads =
+            value_t!(matches, RocksdbCompactionThreadsArg::NAME, NonZeroUsize).map_err(|err| {
+                Box::<dyn std::error::Error>::from(format!(
+                    "failed to parse rocksdb_compaction_threads: {err}"
+                ))
+            })?;
+
+        let rocksdb_flush_threads = value_t!(matches, RocksdbFlushThreadsArg::NAME, NonZeroUsize)
+            .map_err(|err| {
+            Box::<dyn std::error::Error>::from(format!(
+                "failed to parse rocksdb_flush_threads: {err}"
+            ))
```

Comment:
The formatting looks off here but CI is passing 🤔

Reply:
yeah, that’s just how fmt works. they consider it valid formatting haha 🙈. no need to worry about it for now. I’ve removed these code because we don't need to wrap the error